### PR TITLE
feat: support form reset for inputs

### PR DESCRIPTION
**Pull request by ValeraS**

### CODE REVIEW

File: src/components/Checkbox/__tests__/Checkbox.test.tsx [hunk 1]
```diff
@@ -191,4 +191,97 @@ describe('Checkbox', () => {
...
+                    </form>
+                );
+            }
+
+            render(<Test />);
+            const form = screen.getByTestId('form');
+            expect(form).toHaveFormValues({'checkbox-field': true});
+
+            await userEvent.tab();
+            await userEvent.keyboard('{Enter}');
```
Comment (amje) [inline src/components/Checkbox/__tests__/Checkbox.test.tsx:278]: Why don't use just click the checkbox?

File: src/components/controls/TextArea/__tests__/TextArea.test.tsx [hunk 2]
```diff
@@ -152,4 +152,94 @@ describe('TextArea', () => {
...
+            }
+
+            render(<Test />);
+            // eslint-disable-next-line testing-library/no-node-access
+            const inputs = document.querySelectorAll('[name=text-field]');
+            expect(inputs.length).toBe(1);
+            expect(inputs[0]).toHaveValue('value');
+
+            await userEvent.tab();
+            await userEvent.keyboard('text');
```
Comment (amje) [inline src/components/controls/TextArea/__tests__/TextArea.test.tsx:236]: We can use [`type`](https://testing-library.com/docs/user-event/utility#type) here

File: src/components/types.ts [hunk 3]
```diff
@@ -50,18 +50,16 @@ export interface ControlGroupOption<ValueType extends string = string> {
     title?: string;
 }
 
-export interface ControlGroupProps<ValueType extends string = string> {
+export interface ControlGroupProps<ValueType extends string = string> extends AriaLabelingProps {
     name?: string;
-    value?: ValueType;
+    value?: ValueType | null;
```
Comment (amje) [inline src/components/types.ts:55]: `onUpdate` will always be returning the value?

File: src/hooks/private/useRadioGroup/useRadioGroup.ts [hunk 4]
```diff
@@ -39,44 +39,44 @@ export function useRadioGroup<ValueType extends string = string>(
     } = props;
 
     const controlId = useUniqId();
-    const [valueState, setValueState] = React.useState(
-        defaultValue ?? options[0]?.value?.toString(),
+    const [currentValue, setValueState] = useControlledState<string | null, ValueType>(
+        value,
+        // FIXME: Do not set defaultValue to first option value
+        defaultValue ?? options[0]?.value ?? null,
```
Comment (amje) [inline src/hooks/private/useRadioGroup/useRadioGroup.ts:45]: Please submit the issue about it, we will address this in the next major

File: src/hooks/private/useRadioGroup/useRadioGroup.ts [hunk 5]
```diff
@@ -39,44 +39,44 @@ export function useRadioGroup<ValueType extends string = string>(
...
-    const optionsProps = options.map((option) => ({
+    const optionsProps: OptionsProps<ValueType>[] = options.map((option) => ({
         name: name || controlId,
         value: option.value,
         content: option.content,
         title: option.title,
         checked: currentValue === String(option.value),
         disabled: disabled || option.disabled,
         onChange: handleChange,
+        // FIXME: onFocus and onBlur should be on the container
```
Comment (amje) [inline src/hooks/private/useRadioGroup/useRadioGroup.ts:76]: Will it change anything?

Comment (ValeraS) [reply src/hooks/private/useRadioGroup/useRadioGroup.ts:76]: Yes, now onFocus/onBlur is triggered when switching between radio inputs within a group.

File: src/components/Checkbox/__tests__/Checkbox.test.tsx (hunk 1 above)
Comment (ValeraS) [reply src/components/Checkbox/__tests__/Checkbox.test.tsx:278]: why not? This tests that component is accessible from keyboard. And `click` needs element as parameter.

File: src/components/controls/TextArea/__tests__/TextArea.test.tsx (hunk 2 above)
Comment (ValeraS) [reply src/components/controls/TextArea/__tests__/TextArea.test.tsx:236]: `type` needs an input element. `tab` + `keyboard` close to user behavior.

File: src/components/Checkbox/__tests__/Checkbox.test.tsx (hunk 1 above)
Comment (amje) [reply src/components/Checkbox/__tests__/Checkbox.test.tsx:278]: You rely on elements order, and on the fact that the first focusable element is what you need.

File: src/components/controls/TextArea/__tests__/TextArea.test.tsx (hunk 2 above)
Comment (amje) [reply src/components/controls/TextArea/__tests__/TextArea.test.tsx:236]: Replied above

File: src/components/Checkbox/__tests__/Checkbox.test.tsx (hunk 1 above)
Comment (ValeraS) [reply src/components/Checkbox/__tests__/Checkbox.test.tsx:278]: Yes, the same as a user

File: src/hooks/private/useRadioGroup/useRadioGroup.ts (hunk 5 above)
Comment (ValeraS) [reply src/hooks/private/useRadioGroup/useRadioGroup.ts:76]: No if use `useFocusWithin` hook

File: src/hooks/private/useRadioGroup/useRadioGroup.ts (hunk 4 above)
Comment (amje) [reply src/hooks/private/useRadioGroup/useRadioGroup.ts:45]: Created #1663

**Review by amje — APPROVED** [review]

**Pull request merged**